### ISSUE TITLE
Should we expose implicit audio session state

### ISSUE THREAD

**Issue by youennf** [youennf/audio-focus#2]
When playing a HTMLMediaElement, starting an AudioContext or capturing microphone, the User Agent is modifying its internal AudioSession.
Should we expose this AudioSession and its states?
For instance, the implicit AudioSession might change from playback to play-and-record when calling getUserMedia.
Should this audio session type change be exposed?

**Comment by youennf** [comment]
One use case to expose the implicit audio session is to expose the suspended state.
`navigator.audioSession` could be overridden when the session type changes for instance.
Say `getUserMedia` is called, `getUserMedia` algorithm would try to use the current audio session if the type allows it.
If the type is not matching, the algorithm would create a new one AudioSession and inactivating/overriding the current one, thus triggering corresponding events.

**Comment by youennf** [comment]
Ditto with HTMLMediaElement or AudioContext:
- When HTMLMediaElement starts playing audio and audio session is null, a new one is created with type 'playback'. 
- When AudioContext starts playing audio and audio session is null, a new one is created with type 'ambient'.

Otherwise the current audio session is used.

**Comment by youennf** [comment]
The alternative is to introduce a type setter, plus event in case UA has to override/change the audio session type: `audioSession.type = "play-and-record"`

**Comment by jernoble** [comment]
I don't think we should expose the underlying session state; leaving it as "auto" is sufficient.
